gradle/scripts/lib/kotlin.gradle

```
@@ -2,15 +2,21 @@ configure(projectsWithFlags('kotlin')) {
 
     apply plugin: 'kotlin'
 
-    def target = project.findProperty('javaTargetCompatibility') ?: '1.8'
-    def compilerArgs = ['-java-parameters', '-Xjsr305=strict', '-Xskip-prerelease-check']
-    compileKotlin {
-        kotlinOptions.jvmTarget = target
-        kotlinOptions.freeCompilerArgs = compilerArgs
-    }
-    compileTestKotlin {
-        kotlinOptions.jvmTarget = target
-        kotlinOptions.freeCompilerArgs = compilerArgs
+    // compileJmhKotlin is injected by 'jmh' plugin while a benchmark module is being initialized.
+    afterEvaluate {
+        def target = project.tasks.getByName('compileJava').targetCompatibility ?:
```

nit; I know that `java` flag is required now, but eventually I think we may want to allow only `kotlin` projects as well

```suggestion
        def target = project.tasks.findByName('compileJava')?.targetCompatibility ?:
```